pip/commands/list.py

```diff
@@ -72,21 +74,21 @@ def __init__(self, *args, **kw):
                   "pip only finds stable versions."),
         )
 
-        # TODO: When we switch the default, set default=True here.
         cmd_opts.add_option(
-            '--columns',
-            action='store_true',
-            dest='columns',
-            # default=True,
-            help="Align package names and versions into vertical columns.",
+            '--format',
+            action='store',
+            dest='list_format',
+            choices=('legacy', 'columns', 'freeze', 'json'),
+            help="Select the output format among: legacy (default), columns, "
+                 "freeze or json.",
         )
 
         cmd_opts.add_option(
-            '--no-columns',
-            action='store_false',
-            dest='columns',
-            help=("Do not align package names and versions into "
-                  "vertical columns (old-style formatting)"),
+            '--columns',
+            action='store_const',
+            const='columns',
+            dest='list_format',
+            help="Align package names and versions into vertical columns.",
         )
 
```

Comment:
Do we need to retain ```--columns``` if we now have ```--format=columns```? It's not as if ```--columns``` was ever in a released version, so we can simply remove it without any deprecation period.